apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -921,6 +949,7 @@ impl FetchDependencyGraph {
         let mut redundant_edges = HashSet::new();
         self.collect_redundant_edges(node_index, &mut redundant_edges);
 
+        self.on_modification();
```

Comment:
Shouldn't we check `is_empty` here?
```suggestion
        if !redundant_edges.is_empty() {
            self.on_modification();
        }
```